tools/build/src/editor/che-editors-yaml.ts

```diff
@@ -7,34 +7,157 @@
-  repository: string;
-  firstPublicationDate: string;
+interface CheEditorPlatformYaml {
+  inlined?: string;
+  uri?: string;
   endpoints?: CommonEndpointYaml[];
-  containers?: CheEditorContainerYaml[];
-  initContainers?: CheEditorContainerYaml[];
+}
+
+interface CheEditorKubernetesYaml {
+  name: string;
+  namespace?: string;
+}
+
+interface CheEditorPluginYaml {
+  id?: string;
+  kubernetes?: CheEditorKubernetesYaml;
+  uri?: string;
+  commands?: CheEditorCommandYaml[];
+  components?: CheEditorComponentYaml[];
+  registryUrl?: string;
+}
+
+export interface CheEditorVolume {
+  ephemeral?: boolean;
+  size?: string;
+}
+
+export interface CheEditorComponentYaml {
+  container?: CheEditorContainerYaml;
+  kubernetes?: CheEditorPlatformYaml;
+  openshift?: CheEditorPlatformYaml;
+  plugin?: CheEditorPluginYaml;
+  volume?: CheEditorVolume;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  attributes?: { [key: string]: any };
+  name?: string;
+}
+
+interface CheEditorCompositeYaml {
+  commands?: string[];
+  group?: CheEditorGroupYaml;
+  label?: string;
+  parallel?: boolean;
+}
+
+interface CheEditorExecYaml {
+  commandLine: string;
+  component: string;
+  env?: CommonEnvYaml[];
+  group?: CheEditorGroupYaml;
+  hotReloadCapable?: boolean;
+  label?: string;
+  workingDir?: string;
+}
+
+interface CheEditorVscodeObjectYaml {
+  inlined?: string;
+  uri?: string;
+  group: CheEditorGroupYaml;
+}
+
+interface CheEditorCommandYaml {
+  apply?: CheEditorApplyYaml;
+  composite?: CheEditorCompositeYaml;
+  exec?: CheEditorExecYaml;
+  vscodeLaunch?: CheEditorVscodeObjectYaml;
+  vscodeTask?: CheEditorVscodeObjectYaml;
+  attributes?: { [key: string]: string };
+  id?: string;
+}
+
+interface CheEditorGroupYaml {
+  isDefault?: boolean;
+  kind: 'build' | 'run' | 'debug';
+}
+
+interface CheEditorApplyYaml {
+  component: string;
+  group?: CheEditorGroupYaml;
+  label?: string;
+}
+
+interface CheEditorEventYaml {
+  postStart?: string[];
+  postStop?: string[];
+  preStart?: string[];
+  preStop?: string[];
+}
+
+interface CheEditorMetadataYaml {
+  attributes: { [key: string]: string };
+  description?: string;
+  displayName?: string;
+  globalMemoryLimit?: string;
+  icon?: string;
+  name: string;
+  tags?: string[];
+  version?: string;
+}
+
+interface CheEditorGitCheckoutFromYaml {
+  remote?: string;
+  revision?: string;
+}
+
+interface CheEditorVCSYaml {
+  checkoutFrom?: CheEditorGitCheckoutFromYaml;
+  remotes?: { [key: string]: string };
+}
+
+interface CheEditorProjectYaml {
+  git?: CheEditorVCSYaml;
+  gitHub?: CheEditorVCSYaml;
+  zip?: { location: string };
+  attributes?: { [key: string]: string };
+  clonePath?: string;
+  description?: string;
+  name: string;
+  subDir?: string;
+  sparseCheckoutDirs?: string[];
+}
+
+interface CheEditorParentYaml extends CheEditorPluginYaml {
+  projects?: CheEditorProjectYaml[];
+  starterProjects?: CheEditorProjectYaml[];
+}
+
+export interface CheEditorYaml {
```

Comment:
since last week, we now have proper types for devfile API

We should get rid of all *Yaml objects defining the Devfile API as now https://github.com/devfile/api is exposed at https://www.npmjs.com/package/@devfile/api

import { V1alpha2DevWorkspace } from '@devfile/api';

Reply:
This requires quite big refactoring, so I think we can do it in another iteration.

Reply:
:+1: to doing it in another iteration

Reply:
I'm fine to make it as a new iteration